packages/wdio-sync/src/index.js

```diff
@@ -83,23 +84,23 @@ const executeAsync = function (fn, repeatTest = 0, args = []) {
     }
 
     /**
-     * if we don't retry just return result
+     * handle promise response
      */
-    if (repeatTest === 0 || !result || typeof result.catch !== 'function') {
-        return new Promise(resolve => resolve(result))
+    if (result && typeof result.catch === 'function') {
+        return result.catch((e) => {
+            if (repeatTest) {
+                return executeAsync(fn, --repeatTest, args)
+            }
+
+            e.stack = e.stack.split('\n').filter(STACKTRACE_FILTER_FN).join('\n')
+            return Promise.reject(e)
+        })
     }
 
     /**
-     * handle promise response
+     * if we don't retry just return result
      */
-    return result.catch((e) => {
-        if (repeatTest) {
-            return executeAsync(fn, --repeatTest, args)
-        }
-
-        e.stack = e.stack.split('\n').filter(STACKTRACE_FILTER_FN).join('\n')
```

Comment:
Previously I think this was unreachable because if `repeatTest` is falsey then it must be 0 which would be handled in the previous if statement.